python/ray/_private/accelerators/nvidia_gpu.py

```diff
@@ -53,55 +50,82 @@ def get_current_process_visible_accelerator_ids() -> Optional[List[str]]:
 
     @staticmethod
     def get_current_node_num_accelerators() -> int:
-        num_gpus = 0
-        if importlib.util.find_spec("GPUtil"):
-            gpu_list = GPUtil.getGPUs()
-            num_gpus = len(gpu_list)
-        elif sys.platform.startswith("linux"):
-            proc_gpus_path = "/proc/driver/nvidia/gpus"
-            if os.path.isdir(proc_gpus_path):
-                num_gpus = len(os.listdir(proc_gpus_path))
-        elif sys.platform == "win32":
-            props = "AdapterCompatibility"
-            cmdargs = ["WMIC", "PATH", "Win32_VideoController", "GET", props]
-            lines = subprocess.check_output(cmdargs).splitlines()[1:]
-            num_gpus = len([x.rstrip() for x in lines if x.startswith(b"NVIDIA")])
-        return num_gpus
+        try:
+            pynvml.nvmlInit()
+        except pynvml.NVMLError:
+            return 0  # pynvml init failed
+        driver_version = pynvml.nvmlSystemGetDriverVersion()
+        device_count = pynvml.nvmlDeviceGetCount()
+        cuda_devices = []
+        for index in range(device_count):
+            handle = pynvml.nvmlDeviceGetHandleByIndex(index)
+            mig_enabled = os.environ.get(
+                ray_constants.RAY_ENABLE_MIG_DETECTION_ENV_VAR, False
+            )
+            if mig_enabled:
+                try:
+                    max_mig_count = pynvml.nvmlDeviceGetMaxMigDeviceCount(handle)
+                except pynvml.NVMLError_NotSupported:
+                    cuda_devices.append(str(index))
+                    continue
+                for mig_index in range(max_mig_count):
+                    try:
+                        mig_handle = pynvml.nvmlDeviceGetMigDeviceHandleByIndex(
+                            handle, mig_index
+                        )
+                        mig_uuid = ""
+                        if Version(driver_version) >= Version(MIG_UUID_DRIVER_VERSION):
+                            mig_uuid = pynvml.nvmlDeviceGetUUID(mig_handle)
+                        else:
+                            mig_uuid = (
+                                f"MIG-{pynvml.nvmlDeviceGetUUID(handle)}"
+                                f"/{pynvml.nvmlDeviceGetComputeInstanceId(mig_handle)}"
+                                f"/{pynvml.nvmlDeviceGetGpuInstanceId(mig_handle)}"
+                            )
```

Comment:
```suggestion
                            mig_uuid = f"MIG-{pynvml.nvmlDeviceGetUUID(handle)}"
```

It should be either:

```text
MIG-<MIG-UUID>
```

or

```text
MIG-GPU-<GPU-UUID>/<GI ID>/<CI ID>
```